File: WIP/2129-Execution_of_All_Suffix_Instructions_Staying_in_a_Grid.ts
function executeInstructions(
  n: number,
  startPos: number[],
  s: string
): number[] {
  let answer: number[] = [];

  for (let i = 0; i < s.length; i++) {
    let currPos = startPos;
    let goodSteps = 0;

    for (let j = i; j < s.length; j++) {
      switch (s[j]) {
        case "L": {
          currPos[1]--;
          if (currPos[1] < 0) {
            break;
          } else {
            goodSteps++;
            console.log(
              `Incrementing goodSteps by 1 because of "L" matches ${s[j]}, currPos is ${currPos}`
            );
          }
        }

        case "R": {
          currPos[1]++;
          if (currPos[1] > n - 1) {
            break;
          } else {
            goodSteps++;
            console.log(
              `Incrementing goodSteps by 1 because of "R" matches ${s[j]}, currPos is ${currPos}`
            );
          }
        }

        case "U": {
          currPos[0]--;
          if (currPos[0] < 0) {
            break;
          } else {
            goodSteps++;
            console.log(
              `Incrementing goodSteps by 1 because of "U" matches ${s[j]}, currPos is ${currPos}`
            );
          }
        }

        case "D": {
          currPos[0]++;
          if (currPos[0] > n - 1) {
            break;
          } else {
            goodSteps++;
            console.log(
              `Incrementing goodSteps by 1 because of "D" matches ${s[j]}, currPos is ${currPos}`
            );
          }
        }
      }
    }

    break;

    answer.push(goodSteps);
  }

  return answer;
}

console.log(executeInstructions(3, [0, 1], "RRDDLU"));
// Answer:  [ 4, 3, 3, 2, 1, 0 ]
// Expected:  [1, 5, 4, 3, 1, 0]

File: WIP/2129-Execution_of_All_Suffix_Instructions_Staying_in_a_Grid.py
class Solution:
    def executeInstructions(self, n: int, startPos: list[int], s: str) -> list[int]:
        answer = []

        for i in range(len(s)):
            print(f"i is {i}")
            curr_pos = startPos
            good_steps = 0

            for j in range(i, len(s)):
                print(f"\tj is {j}")
                print(
                    f"\t Instruction is {s[j]} and grid is [{curr_pos[0]} x {curr_pos[1]}]"
                )

                match s[j]:
                    case "L":
                        curr_pos[1] -= 1
                        if curr_pos[1] < 0:
                            break
                        else:

                            good_steps += 1

                    case "R":
                        curr_pos[1] += 1
                        if curr_pos[1] > n - 1:
                            break
                        else:
                            good_steps += 1

                    case "U":
                        curr_pos[0] -= 1
                        if curr_pos[0] < 0:
                            break
                        else:
                            good_steps += 1

                    case "D":
                        curr_pos[0] += 1
                        if curr_pos[0] > n - 1:
                            break
                        else:
                            good_steps += 1

            answer.append(good_steps)
            print("\n")

        return answer


solution = Solution()
print(solution.executeInstructions(3, [0, 1], "RRDDLU"))

# Expected answer: [1, 5, 4, 3, 1, 0]
